drivers/entropy/entropy_stm32.c

```diff
@@ -111,6 +112,24 @@ static int entropy_stm32_got_error(RNG_TypeDef *rng)
 	return 0;
 }
 
+#ifdef CONFIG_PM
+static void entropy_stm32_pm_constraint_set(struct entropy_stm32_rng_dev_data *data)
+{
+	if (!data->pm_constraint_on) {
+		data->pm_constraint_on = true;
+		pm_constraint_set(PM_STATE_SUSPEND_TO_IDLE);
+	}
+}
+
+static void entropy_stm32_pm_constraint_release(struct entropy_stm32_rng_dev_data *data)
+{
+	if (data->pm_constraint_on) {
+		data->pm_constraint_on = false;
+		pm_constraint_release(PM_STATE_SUSPEND_TO_IDLE);
+	}
+}
```

Comment:
What is the point of these wrapper functions? If the driver is well-behaved, it will never call a release before a previous set. API is reference counted, so it is fine to call set multiple times as soon as release is later called the same times.

Reply:
I just looked to other implementations like stm32 uart driver. It's more a safeguard indeed to not call the set more than intended

Reply:
This needs a justification, constraint API can be safely called more than once as soon as set/release order is respected.

Reply:
yes, but the pm constraint api is not tracking the count per driver which is calling, so let's say there is a mistake in this driver, it would release the constraint of another driver. I think the safegaurd doesn't hurt, but if you don't agree, it's easy to remove.

Reply:
that's clear, I will rework this

Reply:
done